### PR TITLE
US88714 - fixing description Id

### PR DESCRIPTION
**Pull request by pbrown-d2l**

### CODE REVIEW

File: d2l-date-picker.html [hunk 1]
```diff
@@ -137,6 +141,8 @@
 
 			ready: function() {
 				this._target = this.$.input;
+				this._descriptionId = 'd2l-date-picker-description-' + ( Math.floor( Math.random() * 8999 ) + 1000 );
```
Comment (awikkerink) [inline d2l-date-picker.html]: This should be an incrementing variable.

File: d2l-date-picker.html [hunk 2]
```diff
@@ -137,6 +141,8 @@
...
+				this._target.setAttribute( 'aria-describedby', this._descriptionId );
```
Comment (awikkerink) [inline d2l-date-picker.html]: You can set `aria-describedby` in the template via `aria-describedby$="[[_descriptionId]]"`

File: d2l-date-picker.html (hunk 1 above)
Comment (pbrown-d2l) [inline d2l-date-picker.html]: Is there an example of this in another control somewhere?

File: d2l-date-picker.html (hunk 2 above)
Comment (pbrown-d2l) [inline d2l-date-picker.html]: ah. cool. I tried something similar. We couldn't get it to work. I like your suggestion better.

File: d2l-date-picker.html (hunk 1 above)
Comment (awikkerink) [inline d2l-date-picker.html]: https://github.com/BrightspaceUI/time-picker/blob/master/d2l-time-picker.html#L249

Comment (awikkerink) [inline d2l-date-picker.html]: https://github.com/BrightspaceUI/time-picker/blob/master/d2l-time-picker.html#L112

Comment (pbrown-d2l) [inline d2l-date-picker.html]: Interesting. Thanks I'll add that in

Comment (dlockhart) [inline d2l-date-picker.html]: There's also the `getUniqueId` method on [d2l-polymer-behaviors](https://github.com/Brightspace/d2l-polymer-behaviors-ui/blob/master/d2l-id.html).

File: d2l-date-picker.html [hunk 3]
```diff
@@ -23,7 +23,7 @@
 					display: block;
 				}
 				:host([hide-label]) > label,
-				#d2l-date-picker-description {
+				.d2l-date-picker-description {
```
Comment (dbatiste) [inline d2l-date-picker.html]: Is this meant to be offscreen?  If so, I am curious if our [mixin](https://github.com/BrightspaceUI/offscreen/blob/master/d2l-offscreen-shared-styles.html#L6) can/should be used.

File: d2l-date-picker.html [hunk 4]
```diff
@@ -71,13 +81,22 @@
...
+					on-focus="_handleFocus"
+					aria-describedby$="[[_descriptionId]]"
+				/>
+				<div id="[[_descriptionId]]" class="d2l-date-picker-description" >{{description}}</div>
 			</iron-input>
 		</vaadin-date-picker-light>
 	</template>
 
 	<script>
+		var instances = 0;
```
Comment (awikkerink) [inline d2l-date-picker.html]: This is a global variable. See @dlockhart's comment:

https://github.com/BrightspaceUI/date-picker/pull/20#discussion_r130608488

File: d2l-date-picker.html [hunk 5]
```diff
@@ -18,16 +20,25 @@
...
 				:host([hide-label]) > label,
-				#d2l-date-picker-description {
-					left: -10000px;
-					position: absolute;
+				.d2l-date-picker-description {
+					@apply --d2l-offscreen;
 				}
 
+				:host(:dir(rtl)) :host([hide-label]) > label,
+				.d2l-date-picker-description {
```
Comment (awikkerink) [inline d2l-date-picker.html]: `:host(:dir(rtl)) .d2l-date-picker-description`?

File: d2l-date-picker.html [hunk 6]
```diff
@@ -18,16 +20,25 @@
...
+					@apply --d2l-offscreen-rtl;
+				}
+				:host-context([dir="rtl"]) :host([hide-label]) > label,
+				.d2l-date-picker-description {
```
Comment (awikkerink) [inline d2l-date-picker.html]: `:host-context([dir="rtl"]) .d2l-date-picker-description`

**Comment by pbrown-d2l** [comment]
In order to get the accessibility upgrades to actually work. I will be replacing the d2l-text-input with a native input element like it was previously. Down the road, when the d2l-text-input contains a nice solution for encapsulating a label and a description we can  swap it back in.

**Comment by pbrown-d2l** [comment]
@awikkerink - Are you able to merge this? I don't have permission to do so. Thanks in advance.

**Comment by awikkerink** [comment]
Done and released in 0.0.12

**Pull request merged**